packages/cdktf/lib/terraform-move-addresses.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { IConstruct } from "constructs";
+import { TerraformResource } from "./terraform-resource";
+import { TerraformStack } from "./terraform-stack";
+import { App } from "./app";
+
+/**
+ *
+ */
+export class TerraformResourceAddressMap {
+  private _terraformAddressMap: Map<string, TerraformResource> = new Map<
+    string,
+    TerraformResource
+  >();
+
+  // what do we need?
+  /**
+   * map for a tag -> resource path !!!
+   * function to add to the resource path at a tag
+   * function to return the specificed resource address at a certain tag
+   */
+
+  public add(resource: TerraformResource, tag: string): void {
+    this._terraformAddressMap.set(tag, resource);
```

Comment:
I think we should throw an error if a tag is already set 🤔

Reply:
Included 👍